### DISCUSSION TITLE
Huion H640P not detected on OpenTabletDriver

### DISCUSSION THREAD

**Discussion by JustinSim-s** [OpenTabletDriver/OpenTabletDriver#2265] [Q&A] [answered]
Hello, I was trying to use otd instead of the huion driver, but it seems that otd can't detect my tablet. I'm using windows, and I already uninstalled the huion tablet driver, and used the winusbcleanup to clean stuff up (there's nothing left, too). I saw that I need to send the diagnostic, so here it is, too. 
[OTD_Diagnostics.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/8772978/OTD_Diagnostics.json.txt)
.

**Comment by jamesbt365** [comment, marked as answer]
Huion when they can't make a useless variant of the same tablet for 5 seconds...

Go into tablet > device string reader
- 9580
- 100
- 201
- request string
- also replace 201 with 200 and get that one as well.

**Reply by JustinSim-s** [reply]
Thanks for the reply!
I did it, and it's HUION_T203_211019 (for the 201), and 紀 N῿Ꮨ (200) with another symbol I can't copy. I hope this helps.
![image](https://user-images.githubusercontent.com/106085664/170470923-13d0db79-f9f0-4108-b2a4-d0c578c783f9.png)

**Reply by jamesbt365** [reply]
I can't remember 100% of what was in the diagnostic and it seems to have expired and thus i can't view it anymore so this is a little bit of a guess.
- Download [this](https://github.com/OpenTabletDriver/OpenTabletDriver/files/8778655/H640P.json.txt) file, remove the `.txt` so its just the `.json` at the end
- Make a `Configurations` folder by the executables of OpenTabletDriver (`OpenTabletDriver.UX.Wpf.exe` and `OpenTabletDriver.Daemon.exe`)
- Move the file downloaded within that folder
- Go back into OpenTabletDriver, Tablet > Detect Tablet
- If the tablet detects go into Tablet > Tablet Debugger (if there is a blue dot when moving the pen about this is good)
- From here move your pen to the bottom right of the tablet (ignore the visualizer, could be incorrect here). Get the highest x and y position coordinate and send them here
- Check if both pen buttons go to true when pressed in the debugger they should do this when pressed.
-  Leave the debugger, check if when the tablet buttons are bound to something like a number and save, check if it works.

**Reply by JustinSim-s** [reply]
Thank you for the answers! I did all the stuff, and finally the driver detected my tablet!
I checked all the stuff you mentioned to test (coordinates, pen buttons, and tablet buttons), and both pen buttons and the tablet buttons works.
The coordinate I was able to get (the highest) was 32000 , 20000.
It was a bit janky and sometimes it won't get to the highest position (31997 , 19983 is the coordinate I see the most), but I did get the 32000, 20000 everytime I opened otd and put my pen on the edge of the tablet.

**Reply by jamesbt365** [reply]
Thank you for the verification, this should detect without a configuration file in the next version of OpenTabletDriver
